### PR TITLE
ensure creation of project team many to many when updating projects

### PR DESCRIPTION
**Pull request by macqueen**
part of moving project/team relation to many to many:
- removes unused form
- fixes a case i missed in https://github.com/getsentry/sentry/pull/6136

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by MaxBittker** [comment]
travis is mad

File: src/sentry/api/endpoints/project_details.py [hunk 1]
```diff
@@ -229,6 +231,11 @@ def put(self, request, project):
 
         if changed:
```
Comment (MaxBittker) [inline src/sentry/api/endpoints/project_details.py]: coverage doesn't seem happy about either of these branches, might be worthwhile?

Comment (macqueen) [reply src/sentry/api/endpoints/project_details.py]: mm now it says 100% of diff hit, so i think it's ok?

File: src/sentry/api/endpoints/project_details.py [hunk 2]
```diff
@@ -220,6 +221,7 @@ def put(self, request, project):
                         'detail': ['The new team is not found.']
                     }, status=400
                 )
+            old_team = project.team
```
Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_details.py]: Can we just use `project.team_id` instead of fully resolving `project.team` ? It seems not necessary since you can just use the team id in the query below.

File: src/sentry/web/forms/add_project.py [hunk 3]
```diff
@@ -1,49 +0,0 @@
-from __future__ import absolute_import
```
Comment (mattrobenolt) [inline src/sentry/web/forms/add_project.py]: Why is this being deleted? Seems unrelated.

**Review by mattrobenolt — APPROVED** [review]

File: src/sentry/web/forms/add_project.py (hunk 3 above)
Comment (macqueen) [reply src/sentry/web/forms/add_project.py]: yeah it was removed here https://github.com/getsentry/sentry/commit/d95c3ebcf219b99ff95ea48cda00476d4597d1f8#diff-615ecf8b7cfbfd9bf1f80ed28814ce6bL9
